libraries/omnitruck_helpers.rb

```diff
@@ -0,0 +1,134 @@
+#
+# Copyright (c) 2014-2015, Chef Software, Inc. <legal@chef.io>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative './helpers'
+require 'net/http'
+require 'json'
+
+module ChefIngredientCookbook
+  # The code in this module will eventually go into mixlib-install
+  module OmnitruckHelpers
+    include Helpers
+
+    # TODO: This method should get product name as a parameter and find the
+    # latest version for it.
+    def current_version(product_name)
+      unless %w(chef chefdk).include?(product_name)
+        fail "Unknown product #{product_name}"
+      end
+
+      JSON.parse("/opt/#{product_name}/version-manifest.json")['build_version']
+    end
+
+    # TODO: This method should get product name as a parameter and find the
+    # latest version for it.
+    def latest_available_version(_product_name)
+      latest_metadata = omnitruck_get('/chef/metadata')
+
+      # Extract the relative path from the response from metadata endpoint
+      relative_path = latest_metadata['relpath']
+
+      # Extract the version from the relative path
+      # TODO: support more than Mac OS X
+      relative_path.split('/').last[/^chef-(.*)\.dmg$/, 1]
+    end
+
+    # TODO: This method should get product name as a parameter and find the
+    # latest version for it.
+    def configure_version(version)
+      # Install mixlib-install
+      chef_gem "#{new_resource.product_name}-mixlib-install" do # ~FC009 foodcritic needs an update
+        package_name 'mixlib-install'
+        compile_time true
+      end
+
+      script "install-#{new_resource.product_name}-#{version}" do
+        action :run
+        code lazy {
+          require 'mixlib/install'
+          installer = Mixlib::Install.new(project: new_resource.product_name, version: version)
+          installer.install_command
+        }
+        if new_resource.product_name == 'chef'
+          notifies :run, 'ruby_block[stop chef run]', :immediately
+        end
+      end
```

Comment:
Yeah there is a bit more work to do here. E.g. on windows we should select `:powershell`. I would like to fix these details in a separate PR since this is already very big.